File: solutions/0160. Intersection of Two Linked Lists/0160.js
// Time complexity: O(n)
// Space complexity: O(1)
var getIntersectionNode = function(headA, headB) {
    let a = headA;
    let b = headB;

    while (a !== b) {
        a = a ? a.next : headB;
        b = b ? b.next : headA;
    }
        
    return a;
};

File: solutions/0160. Intersection of Two Linked Lists/0160.cs
// Time complexity: O(n)
// Space complexity: O(1)
public class Solution {
    public ListNode GetIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA;
        ListNode b = headB;

        while (a != b) {
            a = a != null ? a.next : headB;
            b = b != null ? b.next : headA;
        }
        
        return a;
    }
}